prow/apis/prowjobs/v1/types.go

```diff
@@ -920,6 +920,12 @@ type GCSConfiguration struct {
 	// LocalOutputDir specifies a directory where files should be copied INSTEAD of uploading to blob storage.
 	// This option is useful for testing jobs that use the pod-utilities without actually uploading.
 	LocalOutputDir string `json:"local_output_dir,omitempty"`
+	// CompressFileTypes specify file types that should be gzipped prior to upload.
```

Comment:
None other than making installations opt-in to the usage (I am not sure if this will work with `aws` for example), and having a quick way to turn this off if something goes wrong.